### ISSUE TITLE
Linux + nVidia + prime-run + FTBA = black screen (take 2)

### ISSUE THREAD

**Issue by JakeMoe** [FTBTeam/FTB-App#469]
### What Operating System

Linux (Arch)

### App Version

202202111343-da0b2207d4-release

### UI Version

da0b2207d4

### Debug Code

FTB-DBGLUJOVOWUHA

### Describe the bug

App only shows black screen (no content) when using prime-run.  Relogging FTBTeam/FTB-App-Feedback#89, which has been closed, aginst this new version.  The new UI has made things worse, as now nothing launches when I click "Play" to tell me that I've clicked the right spot.

### Steps to reproduce

Unpack app files to folder
Launch installer with "prime-run ~/FTBA/FTBApp"
Get greeted with black content window

### Expected behaviour

Content shows in window

### Screenshots

Without prime-run (which uses the Intel iGPU)
![image](https://user-images.githubusercontent.com/2450407/153662728-6cf9dcdb-41e3-4e2a-9fcc-0fb9839d78a1.png)

With prime-run (which uses the nVidia GPU)
![image](https://user-images.githubusercontent.com/2450407/153662756-73d3d08f-365b-4cd5-b668-9e1cca28f035.png)

### Additional information

As mentioned in the previous ticket, "prime-run" is a tool in Linux on nVidia hybrid graphics systems used to force-start applications using the discrete nVidia GPU rather than the integrated Intel one. It's nothing more than a shell script that sets a few nVidia-specific environment variables and then launches whatever you put as the argument. Here's the whole script:

```
#!/bin/bash
__NV_PRIME_RENDER_OFFLOAD=1 __VK_LAYER_NV_optimus=NVIDIA_only __GLX_VENDOR_LIBRARY_NAME=nvidia "$@"
```

If needed / interested, a little more info can be found here: https://wiki.archlinux.org/title/PRIME. That's the Arch Linux page for it, but the concept (and the tool) applies to other distributions.

**Comment by JakeMoe** [comment]
From what I understand, the recommended way to use hybrid graphics (have I mentioned I miss the days when the CPU was just a CPU?) is Nvidia's PRIME: https://wiki.archlinux.org/title/NVIDIA_Optimus#Available_methods

PRIME itself is an Nvidia technology: https://us.download.nvidia.com/XFree86/Linux-x86_64/510.47.03/README/primerenderoffload.html

After setting my system accordingly, I can run glinfo and glxinfo with and without prime-run and it shows Nvidia and Intel, respectively.  And I can see similar when I run vanilla Minecraft in the F3 screen; Nvidia GeForce vs Mesa Intel Xe.  Other apps work fine with prime-run, such as Steam and Firefox (for better performance with full-screen YouTube).

I'd be happy with a solution that has me run FTBApp normally, but passes prime-run to Minecraft, but I can only find a way to add java parameters, nothing to put "prime-run" in front of the executable, or otherwise pass the three variables in the prime-run script to java.

I take it --disable-gpu-sandbox should be passed to FTBApp?  When I tried that, it didn't seem to make any difference, and I couldn't see any change in output.  But I did note "[main/INFO] [FTB Subprocess]: Electron integration mode"; I don't suppose there's another mode I could try?  As I mentioned in the previous ticket, I think the issue is with Electron; I have similar problems with prime-run with other Electron apps like Skype, Teams, and VS Code, although oddly not with Discord or Atom.

**Comment by MichaelHillcox** [comment]
This is something we'd have to support our side if we wanted to allow custom linux commands to start the game instance. Can you confirm if not using prime allows the app to function normally?

**Comment by JakeMoe** [comment]
Yes, if I don't use prime-run, it works, but then Minecraft uses the Intel card and FPS sucks.  I can set the environment variables separately and then run without prime-run, and I get the same blank window.

```
user@host ~ $ export __NV_PRIME_RENDER_OFFLOAD=1 
user@host ~ $ export __VK_LAYER_NV_optimus=NVIDIA_only
user@host ~ $ export __GLX_VENDOR_LIBRARY_NAME=nvidia
user@host ~ $ ./FTBA/FTBApp
```

It did give me another idea, though; I've read through Nvidia's docs about what those variables are doing, and tried them in various combinations (each separately, and each pair, 6 different commands); none used the Nvidia card except for mixing __NV_PRIME_RENDER_OFFLOAD and __GLX_VENDOR_LIBRARY_NAME above; the docs seem to say it should use NVIDIA_only for __VK_LAYER_NV_optimus by default, so likely having just the two has the same effect as if I'd added the third.

And yes, my previous post said I have the same issue with Skype, Teams, and VS Code, but oddly not with Discord or Atom.  :-)  II did notice that Atom pulled in a separate electron9 package, while VS Code depends on electron13, so it looks like Atom is using an older version?  Maybe it didn't have the problem?  Discord, Skype for Linux, and MS Teams must use bundled Electron versions, as they doesn't depend on any electron package.

**Comment by Jake-E** [comment]
Should be fixed by #815.
